.github/workflows/reusable-build-python-integration-test-workflow.yml

```diff
@@ -0,0 +1,61 @@
+name: Build and Test
+
+on:
+  workflow_call:
+    inputs:
+      sha:
+        required: true
+        type: string
+      ENVIRONMENT:
+        required: true
+        type: string
+    secrets:
+      WALLET_PRIVATE_KEY:
+        required: true
+      RPC_PROVIDER_URL:
+        required: true
+
+jobs:
+  build_and_test:
+    name: Build and Test
+    timeout-minutes: 60
+    runs-on: ubuntu-latest
+    environment: ${{ inputs.ENVIRONMENT }}
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ["3.10", "3.11"]
+    env:
+      WALLET_PRIVATE_KEY: ${{ secrets.WALLET_PRIVATE_KEY }}
+      RPC_PROVIDER_URL: ${{ secrets.RPC_PROVIDER_URL }}
+
+    steps:
+      - name: Check out code
+        uses: actions/checkout@44c2b7a8a4ea60a981eaca3cf939b5f4305c123b # v4.1.5
+        with:
+          ref: ${{ inputs.sha }}
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+          cache: 'pip'  # Caches pip dependencies
+
+      - name: Install dependencies
+        run: |
+          python -m venv .venv
+          source .venv/bin/activate
+          python -m pip install --upgrade pip
+          pip install -r requirements.txt || exit `
+          pip install pytest pytest-cov  # Common testing packages
+        shell: bash
+
+      - name: Run integration tests
+        run: |
+          source .venv/bin/activate 
+          pytest tests/integration/ --cov=./ --cov-report=xml
+
+      - name: Build package
```

Comment:
Question: Do we need this step to build package after testing?

Reply:
Hey @Kavinjsir yes you are right, later we will add an additional steps, for checking the version and publish to PyPI, but I need to test the `dev` -> `main` pipeline before doing it, so I will remove it now and add them again later